packages/ui-toolkit/src/components/molecules/Calendar/MonthCalendar/index.tsx

```diff
@@ -65,54 +110,102 @@ class MonthCalendar extends React.PureComponent<Props, State> {
     );
   }
 
+  // Handle month click event
+  onMonthClick = (index: number) => {
+    const { dateToShow } = this.state;
+    const newDate = new Date(dateToShow.getFullYear(), index, 1);
 
-  onMonthClick = (index:number) => {
-    const dateToShow = this.state.dateToShow;
-    const newDate = new Date(dateToShow.toString());
-    const presentDate = new Date();
-    const presentMonth = presentDate.getMonth();
-
-    if ((index > presentMonth) && (presentDate.getFullYear() === newDate.getFullYear())) {
+    if (this.isMonthDisabled(index)) {
       return;
     }
 
-    newDate.setDate(1);
-    newDate.setMonth(index);
-
     this.props.onDateChange(newDate);
-
     this.setState({
       dateToShow: newDate
     });
   }
 
-
+  // Handle previous year click event
   handlePrevYearClick = () => {
-    const dateToShow = this.state.dateToShow;
-    const newDate = new Date(dateToShow.toString());
+    if (this.isPreviousYearDisabled()) {
```

Comment:
Can we create a constant for the result of `this.isPreviousYearDisabled()` and use that instead of invoking the function in the conditions?